adding raiseEvents info (#6022)
umbraco/UmbracoDocs: 10/umbraco-cms/reference/notifications/contentservice-notifications.md

- It's hacky. There's no good way to suppress events/notifications this way at a higher (scoped) level.
- There's also hard-coded logic to ignore these parameters sometimes which makes it even more inconsistent.
- There are events below services at the repository level that cannot be controlled by this flag.

**What do we use instead?**

We can suppress notifications at the scope level which makes things consistent and will work for all services that use a Scope. Also, there's no required maintenance to make sure that new service methods will also work.

**How to use scopes**:

- Create an explicit scope and call scope.Notifications.Supress().
- The result of Suppress() is IDisposable, so until it is disposed, notifications will not be added to the queue.

[Example](https://github.com/umbraco/Umbraco-CMS/blob/b69afe81f3f6fcd37480b3b0295a62af44ede245/tests/Umbraco.Tests.Integration/Umbraco.Infrastructure/Scoping/SupressNotificationsTests.cs#L35):

```csharp
using (IScope scope = ScopeProvider.CreateScope(autoComplete: true))
using (IDisposable _ = scope.Notifications.Supress())
{
    // TODO: Calls to service methods here will not have notifications
}
```

> Child scope will inherit the parent Scope's notification object which means if a parent scope has notifications suppressed, then so does the child scope. You cannot call Supress() more than once for the same outer scope instance else an exception will be thrown. This is to ensure that you cannot un-suppress notifications at a child level for an outer scope and ensures that suppressing events is a very explicit thing to do.

Author: ```suggestion
Child scope will inherit the parent Scope's notification object which means if a parent scope has notifications suppressed, then so does the child scope. You cannot call Supress() more than once for the same outer scope instance else an exception will be thrown. This ensures that you cannot un-suppress notifications at a child level for an outer scope. It also ensures that suppressing events is an explicit thing to do.
```

> Why would you want to suppress events? The main reason for ever doing this would be performance for bulk operations. The caller will need to be aware that suppressing events will lead to an inconsistent content cache state (if notifications are suppressed for content or media services). This is because notifications are used by things such as NuCache to populate the cmsContentNu table and populate the content caches. They are also used to populate the Examine indexes.

Author: ```suggestion
**Why would one want to suppress events?** The main reason for ever doing this would be performance for bulk operations. The callers hould be aware that suppressing events will lead to an inconsistent content cache state (if notifications are suppressed for content or media services). This is because notifications are used by NuCache to populate the cmsContentNu table and populate the content caches. They are also used to populate the Examine indexes.
```
